### ISSUE TITLE
Lagrange retains window size incorrectly 

### ISSUE THREAD

**Issue by emilis** [skyjake/lagrange#48]
This maybe harder to reproduce. Possible steps:

- v0.9.0 6e1e7d061996c01ea7c3386620b1b52800401715, compiled on Debian bullseye.
- Lagrange was maximised, closed, opened in another desktop.
- The window was wider than the monitor resolution (by ~20px).
- I had to maximize the window and close it twice until it started maximized.
- Note: I keep the sidebar opened all the time.
- Prefs file attached.

[prefs.cfg.txt](https://github.com/skyjake/lagrange/files/5585393/prefs.cfg.txt)

**Comment by skyjake** [comment]
Yeah I have had some challenges with window position retaining. It seems SDL is handling this slightly differently on each platform.

You may find the Lagrange build option `ENABLE_WINDOWPOS_FIX` to be helpful.

**Comment by mrletourneau** [comment]
Can confirm, this happens in OSX as well. Leaving Lagrange open on my external monitor, then unplugging said monitor will change all UI components and gemini page to a really tiny size. The reverse happens when opening Lagrange on my laptop screen, then plugging into my external monitor. Also, dragging the window between screens will trigger this behavior as well.

<img width="1134" alt="Dragged from large screen to small screen" src="https://user-images.githubusercontent.com/1996781/101044898-e07d6400-354d-11eb-918b-e538af920991.png">

**Comment by skyjake** [comment]
@mrletourneau DPI changes are actually a different problem altogether. I just haven't added support for it yet. Thanks for the reminder, though, I'll make a new issue for this (#83).

**Comment by mrletourneau** [comment]
Ah, I guess I misread the issue description! Thanks for cutting the new ticket! Also just want to say that I think Lagrange is a beautiful browser, I am very impressed. Hands down my favorite way to browse gemini capsules 😎

**Comment by skyjake** [comment]
@mrletourneau Rescaling window contents when moving between retina and non-retina monitors has been implemented for v1.2 (issue #83).

**Comment by stacyharper** [comment]
I think I have the same issue on Pinephone (using dwm). It was perfect last time but today it looked like croopy where the virtual keyboard take place. And now after disabling "retain size" (or something like that) it look like this :

![2021-04-07-19-42-30](https://user-images.githubusercontent.com/14213590/113910859-96e1a600-97d9-11eb-9b0b-51b47df055e8.png)

**Comment by skyjake** [comment]
@Eluminae I have little experience with dwm or Pinephone so not sure how to help you. I've understood dwm is expected to handle the window position and sizing, so in Lagrange you should turn off the retain placement setting.

Are you able to determine the specific Lagrange release versions/commits that work or do not work on your device?

**Comment by stacyharper** [comment]
> Are you able to determine the specific Lagrange release versions/commits that work or do not work on your device?

I didn't said that it worked before then it is now broken after upgrades. It worked the first time I openned Lagrange, then after it was broken, after some days or reboots.

I just tested it again and it now works perfectly (with the same localy builded version). I had disabled "retains size" last time as you mentioned it.

edit: it looks like it also work with the 1.3.2 alpine packaged version
